### DISCUSSION TITLE
EMQX support my dev environment? (armhf, armv7l)

### DISCUSSION THREAD

**Discussion by Jena-K** [emqx/emqx#13150] [Q&A]
## is EMQX support my dev environment?
armhf, armv7l
If emqx don't support, please provide guidance on what I should do.
thx

#### Description

* A description of My dev environment *

# cat /etc/os-release 
PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"

# uname -a
Linux USER 5.10.103-v7l+ #1529 SMP Tue Mar 8 12:24:00 GMT 2022 armv7l GNU/Linux

# dpkg --print-architecture
armhf

# lscpu
Architecture:        armv7l
Byte Order:          Little Endian
CPU(s):              4
On-line CPU(s) list: 0-3
Thread(s) per core:  1
Core(s) per socket:  4
Socket(s):           1
Vendor ID:           ARM
Model:               3
Model name:          Cortex-A72
Stepping:            r0p3
CPU max MHz:         1500.0000
CPU min MHz:         600.0000
BogoMIPS:            270.00
Flags:               half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32

*A description of the issue*

root@USER:/home/user# curl -s https://assets.emqx.com/scripts/install-emqx-deb.sh | sudo bash
Detected operating system as raspbian/buster.
Checking for curl...
Detected curl...
Checking for gpg...
Detected gpg...
Running apt-get update... done.
Installing apt-transport-https... done.
Installing /etc/apt/sources.list.d/emqx_emqx.list...done.
Importing gpg key... done.
Running apt-get update... done.

The repository is setup! You can now install packages.
root@USER:/home/user# sudo apt-get install emqx
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package emqx

**Comment by id** [comment]
We do not provide binary packages for armhf and armv7l architecture.
I can suggest to try to build emqx from source https://github.com/emqx/emqx?tab=readme-ov-file#build-from-source on your hardware.
